src/test/java/com/google/sps/tests/UrlRequestTest.java

```diff
@@ -0,0 +1,139 @@
+
+    String actual = UrlRequest.urlQuery(BASE_PATH, EMPTY_PARAMS);
+    String expected = String.format("{\"error\":%s}", responseCode);
+    Assert.assertEquals(expected, actual);
+  }
+
+  @Test
+  public void successfulURlQuery() throws Exception {
+    int responseCode = 200;
+    String content = "hello world";
+    InputStream inputStream = new ByteArrayInputStream(content.getBytes(StandardCharsets.UTF_8.toString()));
+
+    HttpURLConnection mockedConnection = mockUrlConnection(URL_PATH);
+    PowerMockito.when(mockedConnection.getResponseCode()).thenReturn(responseCode);
+    PowerMockito.when(mockedConnection.getInputStream()).thenReturn(inputStream);
+
+    String actual = UrlRequest.urlQuery(BASE_PATH, EMPTY_PARAMS);
+    String expected = content;
+    Assert.assertEquals(expected, actual);
+  }
+
+  @Test
+  public void formatEmptyParamsMap() {
+    String actual = UrlRequest.getParamsString(EMPTY_PARAMS);
+    String expected = EMPTY_STRING;
+    Assert.assertEquals(expected, actual);
+  }
+
+  @Test
+  public void formatOneParam() {
+    Map<String, String> params = new HashMap<>();
+    params.put("hello", "world");
+    String actual = UrlRequest.getParamsString(params);
+    String expected = "hello=world";
+    Assert.assertEquals(expected, actual);
+  }
+
+  @Test
+  public void formatMultipleParams() {
+    Map<String, String> params = new HashMap<>();
+    params.put("hello", "world");
+    params.put("second", "param2");
+    params.put("third", "param3");
+    String response = UrlRequest.getParamsString(params);
+    List<String> actual = Arrays.asList(response.split("&", 0));
+    List<String> expected = Arrays.asList("hello=world", "second=param2", "third=param3");
+    MatcherAssert.assertThat("Array equality without order", actual, containsInAnyOrder(expected.toArray()));
+  }
+
+  @Test
+  public void encodeEmptyString() {
+    String term = EMPTY_STRING;
+    String actual = UrlRequest.encodeTerm(term);
+    String expected = EMPTY_STRING;
+    Assert.assertEquals(expected, actual);
+  }
+
+  @Test
+  public void encodeWord() {
+    String term = "hello";
+    String actual = UrlRequest.encodeTerm(term);
+    String expected = term;
+    Assert.assertEquals(expected, actual);
+  }
+
+  @Test
+  public void encodeWordsAndSpace() {
+    String term = "hello world";
+    String actual = UrlRequest.encodeTerm(term);
+    String expected = "hello+world";
+    Assert.assertEquals(expected, actual);
+  }
+
+  @Test
+  public void encodeSpecialCharacters() {
+    String term = "#$&+,/:;=?@";
+    String actual = UrlRequest.encodeTerm(term);
+    String expected = "%23%24%26%2B%2C%2F%3A%3B%3D%3F%40";
+    Assert.assertEquals(expected, actual);
+  }
+
+  @Test
+  public void encodeNonASCIICharacter() {
+    String term = "😂";
+    String actual = UrlRequest.encodeTerm(term);
+    String expected = "%F0%9F%98%82";
+    Assert.assertEquals(expected, actual);
+  }
+}
```

Comment:
[Nit] add newline

Reply:
There's already a newline after line 139.

Reply:
Or do you mean after line 138?